sdk/log/record.go

```diff
@@ -170,7 +170,11 @@ func (r *Record) Body() log.Value {
 
 // SetBody sets the body of the log record.
 func (r *Record) SetBody(v log.Value) {
-	r.body = v
+	if !r.allowDupKeys {
+		r.body = r.dedupeBodyCollections(v)
```

Comment:
What affect does this have on the performance of `SetBody`?

Reply:
At this point of time we should do the deduplication anyway (to be compliant with the specification).

The performance benefit of not doing (when `allowDupKeys` is `true`) it is pretty irrelevant given we will remove this opt-in performance optimization probably in  `v1.39.0`. Therefore, I do not think it is worth adding a new benchmark. More here: https://github.com/open-telemetry/opentelemetry-go/pull/7002#issuecomment-3077329060

Reply:
We have benchmarks for the same on log record attributes: https://github.com/open-telemetry/opentelemetry-go/pull/6968#issue-3208558377

Reply:
I do not think we should reject this PR because of the benchmark results. The deduplication is required by the spec.

PS. You can consider adding a sub-benchmark(s) when `WithAllowKeyDuplication` option is used.

Reply:
PTAL @MrAlias